app/code/local/Etech/Mydata/Model/Mysql4/Mydata.php

```diff
@@ -12,12 +12,44 @@ public function loadByField() {
         $table = $this->getTable('mydata/profile');
         $table2 = $this->getTable('mydata/user');
         $cond = $this->_getReadAdapter()->quoteInto('profile.user_id = user.user_id', '');
-        // $where = $this->_getReadAdapter()->quoteInto('t1.list_id = "?"',123);
+        //$where = $this->_getReadAdapter()->quoteInto('profile.peofile_id = "?"',45);
         $select = $this->_getReadAdapter()->select()->from(array('profile' => $table))->join(array('user' => $table2), $cond);
         $mydata = $this->_getReadAdapter()->fetchAll($select);
 
-
         return $mydata;
     }
 
+    public function deleteByCondition($id) {
+        $table = $this->getTable('mydata/profile');
+        $table2 = $this->getTable('mydata/user');
+        $where = array();
+        //$where[] =  $this->_getWriteAdapter()->quoteInto('id = ?',$id);
+        $where[] = $this->_getReadAdapter()->quoteInto("user_id = ? ", $id);
+         $select = $this->_getReadAdapter()->select()->from(array('profile' => $table))->join(array('user' => $table2), $cond);
+        $mydata = $this->_getReadAdapter()->fetchAll($select);

```

Comment:
sorry sir for this here a big mistake by me now it is updated here  no requirement of select() and fetchAll() .